src/main/scala/PropertyChecker.scala

```diff
@@ -0,0 +1,408 @@
+        }
+      }
+
+      val maxExclusive:Option[Either[Int, String]] = if (objectNode.path("maxExclusive").path("dateTime").isMissingNode) {
+        if (objectNode.path("maxExclusive").isMissingNode) {
+          None
+        } else {
+          Option(Left(objectNode.get("maxExclusive").asInt()))
+        }
+      } else {
+        if (objectNode.path("maxExclusive").path("dateTime").isMissingNode) {
+          None
+        } else {
+          Option(Right(objectNode.get("maxExclusive").get("dateTime").asText()))
+        }
+      }
+
+     (minInclusive, minExclusive, maxInclusive, maxExclusive) match {
+       case (Some(Left(minI)), Some(Left(minE)), _, _) => throw new MetadataError(s"datatype cannot specify both minimum/minInclusive ($minI) and minExclusive ($minE)")
+       case (Some(Right(minI)), Some(Right(minE)), _, _) => throw new MetadataError(s"datatype cannot specify both minimum/minInclusive ($minI) and minExclusive ($minE)")
+       case (_, _, Some(Left(maxI)), Some(Left(maxE))) => throw new MetadataError(s"datatype cannot specify both maximum/maxInclusive ($maxI) and maxExclusive ($maxE)")
+       case (_, _, Some(Right(maxI)), Some(Right(maxE))) => throw new MetadataError(s"datatype cannot specify both maximum/maxInclusive ($maxI) and maxExclusive ($maxE)")
+       case (Some(Left(minI)), _,Some(Left(maxI)), _) if minI > maxI => throw new MetadataError(s"datatype minInclusive ($minI) cannot be more than maxInclusive ($maxI)")
+       case (Some(Right(minI)), _,Some(Right(maxI)), _) if minI.asInstanceOf[Int] > maxI.asInstanceOf[Int] => throw new MetadataError(s"datatype minInclusive ($minI) cannot be more than maxInclusive ($maxI)")
+       case (Some(Left(minI)), _, _, Some(Left(maxE))) if minI >= maxE => throw new MetadataError(s"datatype minInclusive ($minI) cannot be greater than or equal to maxExclusive ($maxE)")
+       case (Some(Right(minI)), _, _, Some(Right(maxE))) if minI.asInstanceOf[Int] >= maxE.asInstanceOf[Int] => throw new MetadataError(s"datatype minInclusive ($minI) cannot be greater than or equal to maxExclusive ($maxE)")
+       case (_, Some(Left(minE)), _, Some(Left(maxE))) if minE > maxE => throw new MetadataError(s"datatype minExclusive ($minE) cannot be greater than or equal to maxExclusive ($maxE)")
+       case (_, Some(Right(minE)), _, Some(Right(maxE))) if minE.asInstanceOf[Int] > maxE.asInstanceOf[Int] => throw new MetadataError(s"datatype minExclusive ($minE) cannot be greater than or equal to maxExclusive ($maxE)")
+       case (_, Some(Left(minE)), Some(Left(maxI)), _) if minE >= maxI => throw new MetadataError(s"datatype minExclusive ($minE) cannot be greater than maxInclusive ($maxI)")
+       case (_, Some(Right(minE)), Some(Right(maxI)), _) if minE.asInstanceOf[Int] >= maxI.asInstanceOf[Int] => throw new MetadataError(s"datatype minExclusive ($minE) cannot be greater than maxInclusive ($maxI)")
+
+       case (None, None, None, None) => {}
```

Comment:
I'd be more inclined to say `case _ => {}` That matches any other combination which has fallen through the previous case statements.